### ISSUE TITLE
ci-for-labs-ci-cd-pipeline is currently broken

### ISSUE THREAD

**Issue by jacobsee** [rht-labs/labs-ci-cd#204]
The pipeline is failing for all PRs attempted with 
```
TASK [openshift-applier/roles/openshift-applier : Include the pre/post step role] ***
ERROR! 'tmp_dep_dir' is undefined
```
while trying to apply `ci-cd-deployments : nexus`.
@oybed and I looked at it for a while and it doesn't appear to be grabbing the code in the PR, rather just running on `master`... Potential issue with the stash/unstash steps (lines 144, 162) in the Jenkinsfile. Build 8 in Jenkins on S10 is an example of this occurring.

**Comment by jacobsee** [comment]
Justin asked me to put this in - doesn't look like I can make assignments but this should be for @springdo. Let me know if there are any questions about the issue!

**Comment by oybed** [comment]
@jacobsee I think the `tmp_dep_dir` issue is just a side-effect, and I know that as soon as this issue is fixed you will hit another issue - namely a missing dependency for the `openshift-applier` https://github.com/redhat-cop/openshift-applier/issues/47.  My hopes is that we can get the `openshift-applier` issue taken care of ASAP, so hopefully you won't even see it. :-)

**Comment by oybed** [comment]
BTW: just as a FYI, and not meant to distract from this issue - the output you are seeing above re: `requests` will be fixed with this `openshift-applier` issue: https://github.com/redhat-cop/openshift-applier/issues/47

**Comment by oybed** [comment]
@springdo yes, as mentioned earlier, there may still be an issue with that (and it needs to be investigated), but before we go too deep on that, let's get the CI fixed to where it actually picks up the correct source - right now it doesn't.

**Comment by oybed** [comment]
Here's another evidence that it doesn't. My PR #203 changes the `openshift-applier` to `v2.0.0`, but as seen from this output it is still using `v3.9.0` when testing that PR:

```
[labs-ci-cd-ci-for-labs-ci-cd-pipeline] Running shell script
+ ansible-galaxy install -r requirements.yml --roles-path=roles
- extracting openshift-applier to /tmp/workspace/labs-ci-cd/labs-ci-cd-ci-for-labs-ci-cd-pipeline/roles/openshift-applier
- openshift-applier (v3.9.0) was installed successfully
```

**Comment by oybed** [comment]
Further investigating the `tmp_dep_dir`, I found that the [jenkins-slave-ansible](https://github.com/redhat-cop/containers-quickstarts/blob/master/jenkins-slaves/jenkins-slave-ansible/Dockerfile) has the wrong Ansible version. It is using `2.4.x`, while the requirement for anything using the pre/post steps of the openshift-applier requires `>=2.5.x`. A fix should be submitted to the containers-quickstarts repo. (ansible 2.6 is now GA, so maybe go straight to that release) I wrote a new issue for this: https://github.com/redhat-cop/containers-quickstarts/issues/123

However, the main problem of this issue as outlined before is that the incorrect source is used for the CI build, so that still needs to be fixed.

**Comment by oybed** [comment]
Fix for containers-quickstarts: https://github.com/redhat-cop/containers-quickstarts/pull/124

Still need a fix for this repo (Labs CI/CD) for the CI aspect.

**Comment by sherl0cks** [comment]
Thanks @oybed @jacobsee for looking into this. It seems we moved to declarative, but before declarative supported some important features. Specifically, early versions of declarative required that each `stage` run  on a unique slave, the requirement for `stash`. It looks like `1.3` of declarative has resolved this issue - we are currently running `1.2.8`. I'm going to do a plugin update, move to the new `sequential stage` syntax and see what happens.

**Comment by sherl0cks** [comment]
https://github.com/sherl0cks/labs-ci-cd/blob/fix-ci-issue-204/Jenkinsfile resolves the primary issue in this ticket, but it looks like I need to pull in  redhat-cop/containers-quickstarts#124 to send a reasonable PR. Will get to that next
